sky/packages/sky/lib/widgets/switch.dart

```diff
@@ -20,52 +25,124 @@ const double _kThumbRadius = 10.0;
 const double _kSwitchHeight = _kThumbRadius * 2.0;
 const double _kTrackHeight = 14.0;
 const double _kTrackRadius = _kTrackHeight / 2.0;
-const double _kTrackWidth = _kSwitchWidth - (_kThumbRadius - _kTrackRadius) * 2.0;
+const double _kTrackWidth =
+    _kSwitchWidth - (_kThumbRadius - _kTrackRadius) * 2.0;
+const Duration _kCheckDuration = const Duration(milliseconds: 200);
+
+class Switch extends Component {
+  Switch({Key key, this.value, this.onChanged}) : super(key: key);
+
+  bool value;
+  ValueChanged onChanged;
+
+  Widget build() {
+    return new SwitchWrapper(
+        value: value,
+        onChanged: onChanged,
+        thumbColor: Theme.of(this).accentColor);
+  }
+}
 
-class Switch extends Toggleable {
-  // TODO(jackson): Hit-test the switch so that it can respond to both taps and swipe gestures
+// This wrapper class exists only because Switch needs to be a Component in
+// order to get an accent color from a Theme but Components do not know how to
+// host RenderObjects.
+class SwitchWrapper extends OneChildRenderObjectWrapper {
+  SwitchWrapper({Key key, this.value, this.onChanged, this.thumbColor})
+      : super(key: key);
 
-  Switch({
-    Key key,
-    bool value,
-    ValueChanged onChanged
-  }) : super(key: key, value: value, onChanged: onChanged);
+  bool value;
+  ValueChanged onChanged;
+  Color thumbColor;
+
+  RenderSwitch get root => super.root;
+  RenderSwitch createNode() => new RenderSwitch(value, onChanged, thumbColor);
+
+  void syncRenderObject(SwitchWrapper old) {
+    super.syncRenderObject(old);
+    root.value = value;
+    root.thumbColor = thumbColor;
+    root.onChanged = onChanged;
+  }
+}
+
+class RenderSwitch extends RenderProxyBox {
+  RenderSwitch(this._value, this.onChanged, this._thumbColor) {
```

Comment:
We usually use named arguments since its not obvious what order these parameters should go in.